### ISSUE TITLE
TTS can't say `Nemám Čas` correctly

### ISSUE THREAD

**Issue by veksha** [crow-translate/crow-translate#547]
**Steps to reproduce**
1. Select google TTS engine, for example
2. paste text: `Nemám Čas`
3. press play.
4. voice skips letters á and Č

**Environment**

- OS: Windows 11
- Crow Translate version: 2.10.4

**Comment by Shatur** [comment]
Looks like a translation engine issue.

**Comment by veksha** [comment]
broken after PR:
https://github.com/crow-translate/QOnlineTranslator/pull/51

but i think it is a bug not in PR, but in Qt plugins (directshow and wmf).
i see url is passed like: `m_url.toString().utf16()` in those plugins.
but maybe it should be: `m_url.toString(QUrl::FullyEncoded).utf16()`,
because just `toString()` will return percent-DECODED string.

**Comment by Shatur** [comment]
Yeah, makes sense to me. Probably this is why I had separate define for Windows long time ago.
I'm fine with including a workaround in my code until the upstream issue is fixed.

**Comment by veksha** [comment]
@Shatur, can you try to modify directshow and wmf plugins? i don't know how to easily recompile them.

**Comment by Shatur** [comment]
I don't want to bundle custom Qt version, it's very hard to deploy. I would recommend to open an issue in Qt bugtracker and try to find a workaround for Windows without modifying the Qt source code.

**Comment by veksha** [comment]
i think it's better to revert that PR for now. because it breaks a lot. (basically only latin letters are spoken correctly)
"percent twenty" problem will return though.
i see no other way now. no nice workaround.

**Comment by Shatur** [comment]
Sad to hear...
But yeah, let's revert then.

**Comment by veksha** [comment]
"percent twenty" problem is a very strange issue.

i see in Wireshark that Qt's directshow plugin sometimes makes two or three requests simultaneously instead of just one.
one request with `nsplayer...` user-agent header, second one with `nsplayer wmf...` and sometimes buggy third one request with `User-Agent: Mozilla/4.0 (compatible; MSIE 7.0...`. this 3rd request is wrongly percent-encoded for some unknown reason.

bug of Windows or Qt? i don't know.

**Comment by Shatur** [comment]
> bug of Windows or Qt? i don't know.

Could be.
I hope that switch to Qt6 in the future will help. Because Qt Multimedia was rewritten.

**Comment by veksha** [comment]
another idea. instead of passing URLs to qmediaplayer:
- make our own web request
- download spoken .mp3 file to memory/tempfile
- play it with qmediaplayer

is it possible?

**Comment by Shatur** [comment]
It is possible. But I think it's better to switch Qt6 instead of writing workarounds for old Qt version.

**Comment by VolkMilit** [comment]
It's not possible to play from memory with current (Qt5) implementaion of QtMultimedia. This is also blocking supporting Lingva TTS.

**Comment by ValdikSS** [comment]
Windows, Crow-translate 2.10.10
When translating to Russian, Google TTS speaks as if the text has been encoded in cp1251 and then incorrectly interpreted as cp1252.
`Привет` turns to `Ïðèâåò`, and the TTS says the following:
https://github.com/crow-translate/crow-translate/assets/3054729/c3d34e72-feb7-4238-8736-a97f0f3a5afe


Everything is fine on Linux with Crow-translate 2.10.7.

**Comment by Shatur** [comment]
> Everything is fine on Linux with Crow-translate 2.10.7.

Strange, we didn't change anything between these versions.
